### ISSUE TITLE
MetaboliteSpectralMatcher

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#2992]
- add more scoring functions?
- support other file formats? Which are around?

- Is there some overlap with the peptide spec lib code that can be shared/reused?

**Comment by timosachsenberg** [comment]
We should add support for MSP files.

- add https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/FORMAT/MSPGenericFile.h to MetaboliteSpectralMatching
- One would need to check for MSP specific meta values here:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/ID/MetaboliteSpectralMatching.cpp#L543

Maybe would already work.
Thought @pasqoo

**Comment by pasqoo** [comment]
The two following points need to be discussed:

1. There is not a single MSP file format, unfortunately, but several similar ones. Examples of difference could be:
    - fields names (i.e. the synonyms, some files use `Synon`, some `Synonym`, and so on)
    - points delimiters (`;`, `\t`, etc.), even within the x and y info

2. What info should be loaded from the MSP file, and where/how should it be stored in OpenMS? Meaning what classes/objects/members should be used.
I gave a quick look at `MetaboliteSpectralMatching` and I noticed that it uses `.getPrecursors()`, whereas currently `MSPGenericFile` does not use it. It saves the points with a simple `.push_back()` on the `MSSpectrum` instances.

I would be happy to help making the class support more files. I would need to know: the input format, what data to load, and where.

An example of how `MSPGenericFile`-loaded data is used can be found in class `TargetedSpectraExtractor`: https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp#L492

And here some examples of currently supported formats: https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/FORMAT/MSPGenericFile.h#L45
